lib/utils/timetable.test.ts

```diff
@@ -451,66 +451,61 @@ describe("resolveDevelopmentPlanCSV", () => {
   });
 });
 
-describe("getStageProgress", () => {
-  test("returns IN PROGRESS when startDate is current date", () => {
-    const startDate = new Date();
-
-    const progress = getStageProgress(startDate.toISOString());
-
-    expect(progress).toBe("IN PROGRESS");
-  });
-
-  test("returns NOT STARTED when startDate is in the future", () => {
-    const startDate = new Date();
-
-    startDate.setMonth(startDate.getMonth() + 1);
-
-    const progress = getStageProgress(startDate.toISOString());
-
-    expect(progress).toBe("NOT STARTED");
-  });
-
-  test("returns FINISHED when startDate is in the past and there is no endDate", () => {
-    const startDate = new Date();
-
-    startDate.setMonth(startDate.getMonth() - 1);
-
-    const progress = getStageProgress(startDate.toISOString());
-
-    expect(progress).toBe("FINISHED");
-  });
-
-  test("returns IN PROGRESS when startDate is in the past and endDate is in the future date", () => {
-    const startDate = new Date();
-
-    startDate.setMonth(startDate.getMonth() - 1);
-
-    const endDate = new Date();
-
-    endDate.setMonth(startDate.getMonth() + 1);
-
-    const progress = getStageProgress(
-      startDate.toISOString(),
-      endDate.toISOString()
-    );
-
-    expect(progress).toBe("IN PROGRESS");
-  });
-
-  test("returns FINISHED when endDate date is in the past", () => {
-    const startDate = new Date();
-
-    startDate.setMonth(startDate.getMonth() - 2);
-
-    const endDate = new Date();
-
-    endDate.setMonth(startDate.getMonth() - 2);
+interface StageProgressTestCase {
+  lastUpdatedDate: string;
+  startDate: string;
+  endDate?: string;
+  expectedProgress: string;
+}
 
-    const progress = getStageProgress(
-      startDate.toISOString(),
-      endDate.toISOString()
-    );
+const stageProgressTestCases: StageProgressTestCase[] = [
+  {
+    lastUpdatedDate: "2024-02-07T09:03:53.514Z",
+    startDate: "2024-03-02",
+    expectedProgress: "NOT STARTED",
+  },
```

Comment:
To be honest, I'm not sure what "IN PROGRESS" would mean when there's no end date, in that case it's either done or it's not, so maybe the logic should be:
- `startDate` is before or equal to `lastUpdatedDate` -> "FINISHED"
- `startDate` is after `lastUpdatedDate` -> "NOT STARTED" (but maybe we check with design to see if there's a better name for this)

I think Ollie and I discussed before that while we're capturing the time in the timetable updated event, we don't _have_ to use it for calculating the statuses. So maybe when we calculate statuses, we should ignore the time part of the date? I'll have a look into that and talk with the designers about it (as well as the names and behaviours for the statuses without end dates above).

Reply:
Also, some further thinking, this function currently doesn't know the difference between "there's no `endEventDate` because there's no end event" (i.e. it's not a stage) and "there's no `endEventDate` because the user didn't provide one"